Random fails in antlr

`antlr` fails randomly with the following revision:
```
binding: 9347c94fa8f51f239248d5a7de99ca8cf11e3eb0
mmtk-core: 66edca535ed3a4c5db399d9b4d2fe289fce84360
```

It was executed with the following command lines.
```
/home/gitlab-runner/actions-runner/_work/mmtk-core/mmtk-core/ci-perf-kit/running/build/jdk-mmtk-branch-semispace/jdk/bin/java -Xms144M -Xmx144M  -XX:MetaspaceSize=100M -server -XX:-TieredCompilation -XX:+UseThirdPartyHeap -Dprobes=MMTk -cp /home/gitlab-runner/actions-runner/_work/mmtk-core/mmtk-core/ci-perf-kit/running/../probes/probes.jar:/home/gitlab-runner/actions-runner/_work/mmtk-core/mmtk-core/ci-perf-kit/running/benchmarks/dacapo/dacapo-2006-10-MR2.jar Harness -converge antlr
```

Using `converge` for running dacapo makes it run more iterations, which seems to trigger the problem more frequent. It was as bad as 9 fails out of 40 invocations. We don't know if the problem is within mmtk-core or the openjdk binding. 

The errors include:

1. 
```
panic: Error reading importVocab file 'JavaTokenTypes.txt'
```

2. 
```
/tmp/runbms-gitlab-runner/./scratch/TinyBasicParser.java.antlr.tmp (Too many open files)
Running antlr on grammar antlr/java/TinyBasicTreeWalker.g
ANTLR Parser Generator   Version 2.7.2   1989-2003 jGuru.com
error: file ./scratch/antlr/java/TinyBasicTreeWalker.g not found
Running antlr on grammar antlr/java/tinyc.g
ANTLR Parser Generator   Version 2.7.2   1989-2003 jGuru.com
error: file ./scratch/antlr/java/tinyc.g not found
Running antlr on grammar antlr/java/track.g
ANTLR Parser Generator   Version 2.7.2   1989-2003 jGuru.com
error: file ./scratch/antlr/java/track.g not found
Running antlr on grammar antlr/java/treewalk.g
ANTLR Parser Generator   Version 2.7.2   1989-2003 jGuru.com
error: file ./scratch/antlr/java/treewalk.g not found
Running antlr on grammar antlr/java/unicode.g
ANTLR Parser Generator   Version 2.7.2   1989-2003 jGuru.com
error: file ./scratch/antlr/java/unicode.g not found
Running antlr on grammar antlr/java/xml.g
ANTLR Parser Generator   Version 2.7.2   1989-2003 jGuru.com
error: file ./scratch/antlr/java/xml.g not found
Digest validation failed for stderr.log, expecting 0x459b3713364a15cb610ce084bd22f352833c3255 found 0x2acd9ee96b31aae4be38ac59388d1e31a876a306
===== DaCapo antlr FAILED =====
Validation FAILED for antlr default
```

--- a/openjdk/mmtkHeap.cpp
+++ b/openjdk/mmtkHeap.cpp
@@ -414,8 +414,7 @@ void MMTkHeap::scan_weak_processor_roots(OopClosure& cl) {
 void MMTkHeap::scan_vm_thread_roots(OopClosure& cl) {
    ResourceMark rm;
    MMTkRootScanWorkScope<> root_scan_work(&_num_root_scan_tasks);
-   CodeBlobToOopClosure cb_cl(&cl, false);
-   VMThread::vm_thread()->oops_do(&cl, &cb_cl);
+   VMThread::vm_thread()->oops_do(&cl, NULL);
 }
 
 void MMTkHeap::scan_global_roots(OopClosure& cl) {

--- a/openjdk/mmtkHeap.cpp
+++ b/openjdk/mmtkHeap.cpp
@@ -448,9 +448,7 @@ void MMTkHeap::scan_global_roots(OopClosure& cl) {
 void MMTkHeap::scan_thread_roots(OopClosure& cl) {
    ResourceMark rm;
    MMTkRootScanWorkScope<> root_scan_work(&_num_root_scan_tasks);
-
-   CodeBlobToOopClosure cb_cl(&cl, false);
-   Threads::possibly_parallel_oops_do(false, &cl, &cb_cl);
+   Threads::possibly_parallel_oops_do(false, &cl, NULL);
 }
 
 void MMTkHeap::scan_roots(OopClosure& cl) {

--- a/openjdk/mmtkUpcalls.cpp
+++ b/openjdk/mmtkUpcalls.cpp
@@ -164,8 +164,7 @@ static void mmtk_scan_thread_root(ProcessEdgesFn process_edges, void* tls) {
     ResourceMark rm;
     JavaThread* thread = (JavaThread*) tls;
     MMTkRootsClosure2 cl(process_edges);
-    CodeBlobToOopClosure cb_cl(&cl, false);
-    thread->oops_do(&cl, &cb_cl);
+    thread->oops_do(&cl, NULL);
 }
 
 static void mmtk_scan_object(void* trace, void* object, void* tls) {

